ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuation.kt

```diff
@@ -0,0 +1,113 @@
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        val chainOperator = node.findChildByType(ElementType.SAFE_ACCESS) ?: node.findChildByType(ElementType.DOT)
+        chainOperator ?: return
+        val rightExpression = chainOperator.nextCodeSibling()?.firstChildLeafOrSelf() ?: return
+        val leftExpressionEndBrace =
+            chainOperator.prevLeaf { !it.isWhiteSpace() && it.elementType != ElementType.EXCLEXCL }
+                ?.lastChildLeafOrSelf() ?: return
+        val leftExpressionBeforeEndBrace = leftExpressionEndBrace.prevCodeSibling()?.lastChildLeafOrSelf() ?: return
+        val isLeftExpressionEndBraceInSeparateLine =
+            leftExpressionEndBrace.elementType == ElementType.RBRACE &&
+                hasNewLineInClosedRange(leftExpressionBeforeEndBrace, leftExpressionEndBrace)
+        if (!noNewLineInOpenRange(chainOperator, rightExpression)) {
+            emit(
+                chainOperator.textRange.endOffset,
+                "${
+                    if (isLeftExpressionEndBraceInSeparateLine) {
+                        leftExpressionEndBrace.leavesIncludingSelf()
+                            .takeWhile { it != chainOperator && !it.isWhiteSpace() }.map { it.text }
+                            .joinToString(separator = "")
+                    } else {
+                        ""
+                    }
+                }${chainOperator.text} must merge at the start of next call",
+                true,
+            )
+            if (autoCorrect) {
+                if (isLeftExpressionEndBraceInSeparateLine) {
+                    /*
+                        Detects code like below
+                        bar {
+                            ...
+                        }.
+                        foo() // this should align with previous line }
+                     */
+                    chainOperator.upsertWhitespaceAfterMe("")
+                } else {
+                    chainOperator.upsertWhitespaceBeforeMe(rightExpression.indent())
+                    rightExpression.upsertWhitespaceBeforeMe("")
+                }
+            }
+        } else if (leftExpressionEndBrace.elementType == ElementType.RBRACE &&
+            isLeftExpressionEndBraceInSeparateLine &&
+            !noNewLineInOpenRange(
+                leftExpressionEndBrace,
+                chainOperator,
+            )
+        ) {
+            /*
+                Detects code like below
+                bar {
+                    ...
+                }
+                .foo() // this should align with previous line }
+             */
+            emit(
+                chainOperator.textRange.endOffset,
+                "${chainOperator.text} must must merge at the end of previous call",
+                true,
+            )
+            if (autoCorrect) {
+                chainOperator.upsertWhitespaceBeforeMe("")
```

Comment:
This leaves an empty PsiWhitespace elements in the AST. Technically that is not a problem. But I think that removing an unwanted whitespace element is more clear.

Reply:
Updated with `nextMultilineWhiteSpace.treeParent.removeChild(nextMultilineWhiteSpace)`